src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -80,11 +85,19 @@ public abstract class JsonPropertyInfo
         public Action<object, object?>? Set
         {
             get => _untypedSet;
-            set => SetSetter(value);
+            set
+            {
+                VerifyMutable();
+                SetSetter(value);
+                // Invalidate any JsonIgnore configuration if delegate set manually by user
+                IgnoreNullTokensOnRead = false;
```

Comment:
I'm not sure, why does this need to influence this? we currently don't have any metadata to override this - or is this assuming that IgnoreNullTokensOnRead check is part of the setter?

Reply:
It complements `Set`, similar to how `IgnoreDefaultValuesOnWrite` complements `ShouldSerialize`. A manual user setting should invalidate that configuration.